static/js/elementContentCleaner.js

```diff
@@ -16,47 +16,51 @@ elementContentCleaner.prototype.deleteElement = function() {
 
   var lineToSelect;
   var lineIsHeading = utils.lineIsHeading(currentLine, this.attributeManager);
+
   if (lineIsHeading) {
-    var $lineToDelete = this._getParentSMOfHeading(currentLine);
-    var sceneMarkLineId = $lineToDelete.attr('id');
-    removeSceneMark(sceneMarkLineId);
-    var titleAndSummaryLines = 2; // 2 lines
-    // currentLine now is the number of the line after the deleted HEADING
-    lineToSelect = currentLine - titleAndSummaryLines;
+    // calculate the number of scene mark lines that will be deleted
+    var titleAndSummaryLinesToDelete = this._calculateLinesToDelete(currentLine);
+
+    // calculate the line number to select
+    lineToSelect = currentLine - titleAndSummaryLinesToDelete;
+
+    // delete the heading
+    this._removeElement(currentLine);
+
+    // delete the related scene marks
+    triggerEventRemoveSceneMark([currentLine], true);
```

Comment:
nós esperamos que o heading e SM seja inseparável sempre. Temos inclusive alguns mecanismo que checa isso. Talvez seja mais seguro mudar a `_removeElement` para remover tudo de uma vez só. A única coisa que mudaria seria a função `var intervalToRemove = this._getIntervalToRemove(lineNumberOfSE);` que deveria retornar todo o intervalo. Poderíamos passar aqui uma flag para `removeElement` ou a própria função `removeElement` poderia checar 

``` js
if (heading) {
 // find heading and scene mark intervals to remove
}
```